internal/services/pam/pam.go

```diff
@@ -274,12 +278,14 @@ func (s Service) SetDefaultBrokerForUser(ctx context.Context, req *authd.SDBFURe
 		return nil, status.Error(codes.InvalidArgument, "no user name given")
 	}
 
-	if err = s.brokerManager.SetDefaultBrokerForUser(req.GetBrokerId(), req.GetUsername()); err != nil {
-		return &authd.Empty{}, err
+	// Don't allow setting the default broker to the local broker, because the decision to use the local broker should
+	// be made each time the user tries to log in, based on whether the user is provided by any other NSS service.
+	if req.GetBrokerId() == brokers.LocalBrokerName {
+		return nil, status.Error(codes.InvalidArgument, "can't set local broker as default")
```

Comment:
Maybe we should just return nil instead of an error here? I don't think we should show an error in PAM if we try to set the default broker as the local one. Do we have a special case for this in the PAM module itself to not get here?

Reply:
> Maybe we should just return nil instead of an error here?

I did that at first but changed it to return an error because I prefer the API to not silently ignore a request to set the local broker as the default. It should be made clear to the clients that this is not supported and therefore the clients should ensure that they don't call `SetDefaultBrokerForUser` for the local broker.

The PAM module calls `SetDefaultBrokerForUser` [in the `AcctMgmt` function](https://github.com/ubuntu/authd/blob/5fe9179fa183b93722f9bbc2d1dcf299e31720e3/pam/pam.go#L434-L434). I expect that this function is not called with the broker ID set to the local broker, but I'm not 100% sure, so let's ask @3v1n0. If there is a chance that it is called for the local broker, lets add a check there to avoid calling `SetDefaultBrokerForUser` for the local broker.